### PR TITLE
Supporting Different Kernel JSONs for Various Platforms

### LINKED ISSUE #7515: Powershell Kernel Hangs on Kernel Change for Mac/Linux
In the poweshell kernel's kernel.json, the following line needs to be different for Windows vs. other platforms:
Windows:
```"powershell_command": "powershell"```

Other platforms
```"powershell_command": "pwsh"```

Alternatively, we can just run the install command as @corivera previously suggested.

@corivera any preference?

### PR DESCRIPTION
**Pull request by chlafreniere**
(hit enter too quickly 😄)

Fixes #7515 

Problem: For some kernels (e.g. powershell now, maybe more in the future), there is platform-specific kernel.json required.

Today, we only copy over *the* kernel.json for a kernel instead of having any logic for platform-specific kernel.json's.

I'm sure people will have lots of ideas for how to do this,  so just getting this out there to gather feedback 😄

### CODE REVIEW

File: extensions/notebook/src/jupyter/serverInstance.ts [hunk 1]
```diff
@@ -207,12 +210,36 @@ export class PerNotebookServerInstance implements IServerInstance {
...
+			let realFolder = subFolders[i];
+			// Ignore any system files
+			if (!realFolder.startsWith('.')) {
+				if (realFolder.includes(process.platform)) {
+					let splitFolder = realFolder.split('_' + process.platform);
+					if (splitFolder && splitFolder.length > 0) {
+						realFolder = splitFolder[0];
+					}
+				}
+				await this.utils.copy(path.join(kernelsExtensionSource, subFolders[i]), path.join(this._systemJupyterDir, realFolder));
```
Comment (aaomidi) [inline extensions/notebook/src/jupyter/serverInstance.ts]: What happens if this fails?

Comment (chlafreniere) [reply extensions/notebook/src/jupyter/serverInstance.ts]: Good question. An error message pops up to users here. Any other suggestions?

Comment (aaomidi) [reply extensions/notebook/src/jupyter/serverInstance.ts]: Ah as long as its handled somewhere :) Sometimes these failures completely break the extension so just checking :)

File: extensions/notebook/src/jupyter/serverInstance.ts [hunk 2]
```diff
@@ -207,12 +210,36 @@ export class PerNotebookServerInstance implements IServerInstance {
...
+		const Win32Postfix = '_win32';
 		let kernelsExtensionSource = path.join(this.options.install.extensionPath, 'kernels');
 		this._systemJupyterDir = path.join(this.getSystemJupyterHomeDir(), 'kernels');
 		if (!(await this.utils.exists(this._systemJupyterDir))) {
 			await this.utils.mkDir(this._systemJupyterDir, this.options.install.outputChannel);
 		}
-		await this.utils.copy(kernelsExtensionSource, this._systemJupyterDir);
+		let platform = process.platform;
+		let subFolders = await this.utils.readDir(kernelsExtensionSource);
+		if (platform === 'win32') {
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/jupyter/serverInstance.ts]: Often stuff like this is handled with having the config in the package.json - that seems like a much cleaner way of handling this I would think

File: extensions/notebook/src/jupyter/serverInstance.ts [hunk 3]
```diff
@@ -70,6 +70,9 @@ export class ServerInstanceUtils {
 			return false;
 		}
 	}
+	public async readDir(path: string): Promise<string[]> {
```
[the remark this answers is not in the document]
Comment (aaomidi) [reply extensions/notebook/src/jupyter/serverInstance.ts]: There's an argument to be made to have it be async: https://palantir.github.io/tslint/rules/promise-function-async/

[the remark this answers is not in the document]
Comment (Charles-Gagnon) [reply extensions/notebook/src/jupyter/serverInstance.ts]: Oh interesting, yeah I hadn't considered that. I like it! Ignore my comment then.

File: extensions/notebook/src/jupyter/serverInstance.ts (hunk 2 above)
Comment (chlafreniere) [reply extensions/notebook/src/jupyter/serverInstance.ts]: So there's a much better way to do this and with some code from the powershell extension we'll be setting the powershell_executable property at runtime...

So this functionality isn't needed (and if we need it in the future, I like your idea of adding to package.json). So yeah.

**Pull request merged**